common/namespace/nsregistry/registry.go

```diff
@@ -414,22 +397,23 @@ func (r *registry) refreshNamespaces(ctx context.Context) error {
 	}
 
 	// Make a copy of the existing namespace cache (excluding deleted), so we can calculate diff and do "compare and swap".
```

Comment:
```suggestion
	// Make a copy of the existing namespace maps (excluding deleted), so we can calculate diff and atomically swap.
```
it looks like we don't do compare-and-swap, we just assume only one thread is ever updating

Reply:
done